src/main/java/com/altenheim/kalender/controller/viewController/PopupViewsController.java

```diff
@@ -79,4 +83,34 @@ public String showPasswordInputDialog()
 
         return password.getText();
     }
+
+    public void importDialog(IImportController importController, IEntryFactory entryFactory, Window stage)
+    {        
+        var filePicker = new FileChooser();
+        var file = filePicker.showOpenDialog(stage);
+        if (file == null)
+            return;
+        var importedCalendar = importController.importFile(file.getAbsolutePath());
+        entryFactory.addCalendarToView(importedCalendar);      
+    }
+
+    public void exportDialog(IExportController exportController, ICalendarEntriesModel allEntries, Window stage)
+    {
+        var calendars = allEntries.getAllCalendars();
+        var directoryChooser = new DirectoryChooser();
+        var path = directoryChooser.showDialog(stage);
+        if (path == null)
+            return;
+        for (var calendar: calendars)
+        {
+            try 
```

Comment:
Try in einer For-Schleife ist Performancemäßig nicht schön

Reply:
Ähhhmm. Ja. Der Fehler muss abgefangen werden. Alternative? Die Exceptions einfach an die Methode klatschen so das dann der Aufrufer die ganze Methode in den Catch blockt packt? Oder gleich bis in höchste Ebenen weitergeben? Fält ja dann nicht auf das fünf Ebenen drunter die Schleife (und tausend andere Sachen) im Try Block laufen. Fehler werden da angefangen wo sie entstehen können. Natürlich soll man nicht unnötig nen try block in die Schleife bauen aber es gibt wie so oft kein schwarz und weiß. Schlag ne Alternative vor wenn dir eine einfällt :-)